runtime/bcutil/cfreader.c

```diff
@@ -2056,6 +2058,35 @@ checkAttributes(J9CfrClassFile* classfile, J9CfrAttribute** attributes, U_32 att
 						errorCode = J9NLS_CFR_ERR_BOOTSTRAP_METHODHANDLE__ID;
 						goto _errorFound;
 					}
+
+					numberOfBootstrapArguments = bsm->numberOfBootstrapArguments;
+					for (k = 0; k < numberOfBootstrapArguments; k++) {
+						U_8 cpValueTag = 0;
+						value = bsm->bootstrapArguments[k];
+
+						if((0 == value) || (value > cpCount)) {
+							errorCode = J9NLS_CFR_ERR_BAD_INDEX__ID;
+							goto _errorFound;
+						}
+						/* Validate the constant_pool indexes stored in the bootstrap_arguments array.
+						 * Note: The constant_pool entry at that index must be a CONSTANT_String_info,
+						 * CONSTANT_Class_info, CONSTANT_Integer_info, CONSTANT_Long_info, CONSTANT_Float_info,
+						 * CONSTANT_Double_info, CONSTANT_MethodHandle_info, or CONSTANT_MethodType_info structure.
+						 */
+						cpValueTag = cpBase[value].tag;
+						if ((CFR_CONSTANT_String != cpValueTag)
+						&& (CFR_CONSTANT_Class != cpValueTag)
+						&& (CFR_CONSTANT_Integer != cpValueTag)
+						&& (CFR_CONSTANT_Long != cpValueTag)
+						&& (CFR_CONSTANT_Float != cpValueTag)
+						&& (CFR_CONSTANT_Double != cpValueTag)
+						&& (CFR_CONSTANT_MethodHandle != cpValueTag)
+						&& (CFR_CONSTANT_MethodType != cpValueTag)
```

Comment:
Also, for JDK11, what about Constant_Dynamic?  It should be allowed as a BSM arg as well

Reply:
Already double-checked the Spec at http://openjdk.java.net/jeps/309 and added Constant_Dynamic as valid structure.